src/main/java/shop/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,112 @@
+package shop.service.impl;
+
+import java.util.HashSet;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import shop.dto.request.shopping.cart.AddBookToCartRequestDto;
+import shop.dto.request.shopping.cart.UpdateQuantityBookRequestDto;
+import shop.dto.responce.cart.item.CartItemResponseDto;
+import shop.dto.responce.shopping.cart.ShoppingCartResponseDto;
+import shop.entity.CartItem;
+import shop.entity.ShoppingCart;
+import shop.exception.EntityNotFoundException;
+import shop.mapper.CartItemMapper;
+import shop.mapper.ShoppingCartMapper;
+import shop.repository.BookRepository;
+import shop.repository.CartItemRepository;
+import shop.repository.ShoppingCartRepository;
+import shop.service.ShoppingCartService;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+
+    @Transactional
+    @Override
+    public ShoppingCartResponseDto getShoppingCartById(Long id) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartById(id)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can`t get shopping cart by id: " + id));
+        Set<CartItemResponseDto> cartItemResponseDtos = cartItemMapper
+                .toCartItemResponseDtos(shoppingCart.getCartItems());
+        ShoppingCartResponseDto shoppingCartResponseDto = shoppingCartMapper
+                .toShoppingCartResponseDto(shoppingCart);
+        shoppingCartResponseDto.setCartItems(cartItemResponseDtos);
+        return shoppingCartResponseDto;
```

Comment:
Example of implementation

```suggestion
        return shoppingCartMapper.toDto(shoppingCartRepository.findByUserId(userId));
```